### PR TITLE
Isolate each REPL instance's input buffer via new `sessionId` tag

### PR DESCRIPTION
**Pull request by jmfirth**
This change expands the `input` buffer, creating one for each session.  Sessions are tagged by a `sessionId` and passed around with the sessions's `readline` instance, where necessary.  Addresses #32.

### CODE REVIEW

**Comment by jmfirth** [comment]
I haven't entirely thought it through yet, but `createSession` method may be a great place to consolidate some of the REPL behavior, either by passing in the required input/output streams or the already-created `readline` interface.

**Comment by anmonteiro** [comment]
seems like an object would be more appropriate for holding the sessions. You can still have numbers as the keys in the object.

File: src/js/repl.js [hunk 1]
```diff
@@ -209,19 +216,31 @@ function highlight(rlInt: readline$Interface,
...
   if (!isPasting) {
     // $FlowIssue: these properties exist
-    highlight(rl, c, rl.line, rl.cursor);
+    highlight(sessionId, rl, c, rl.line, rl.cursor);
   }
 }
 
+export function createSession(): number {
+  sessionCount += 1;
+  const sessionId = sessionCount - 1; // start at 0
```
Comment (anmonteiro) [inline src/js/repl.js]: this arithmetic would be pointless to make if we stored the sessions in an object.

**Comment by jmfirth** [comment]
Reverted my changes and isolated just the changes for this issue.  Have not yet addressed:
> All the code path above is doing too many unnecessary accesses to input[sessionId] when we could be storing it in intermediate variables at some places.

File: src/js/repl.js [hunk 2]
```diff
@@ -83,6 +83,21 @@ function consumeBuffer(buffer: string[], stream: stream$Writable | tty$WriteStre
...
 
+export function deleteSession(session: REPLSession): void {
+  delete sessions[session.sessionId];
+}
+
+function stopREPL(): void {
+  socketServerClose();
+
+  const keys = Object.keys(sessions);
+  keys.forEach((sessionId: string) => deleteSession(sessions[parseInt(sessionId)]));
```
Comment (anmonteiro) [inline src/js/repl.js]: `parseInt` is unnecessary, property access can be string

**Pull request merged**